axe.d.ts

```diff
@@ -318,17 +318,17 @@ declare namespace axe {
     open: (topicHandler: TopicHandler) => Close | void;
     post: (
       frameWindow: Window,
-      data: TopicData | ReplyData,
+      data: TopicData,
       replyHandler: ReplyHandler
-    ) => void;
+    ) => boolean | void;
   };
   type Close = Function;
-  type TopicHandler = (data: TopicData, responder?: Responder) => void;
+  type TopicHandler = (data: TopicData, responder: Responder) => void;
   type ReplyHandler = (data: ReplyData, responder?: Responder) => void;
```

Comment:
The `ReplyHandler` type here (and corresponding docs) is still inconsistent with actual usage: https://github.com/dequelabs/axe-core/blob/b388eb3425817edc0742e25542be8d29f7aebdde/lib/core/utils/frame-messenger/message-handler.js#L67

Presuming you want to avoid breaking plugin compat, this should be 

```suggestion
  type ReplyHandler = (message: any | Error, keepalive: boolean, responder: Responder) => void;
```

For the sake of plugin compat, I think it's probably best to make the parameters non-optional; frameMessengers should always be sending them, even if axe-core's default reply handlers ignore them

Reply:
(Given this change, it might be more clear to eliminate the `ReplyData` type and fold it into `TopicData`, since that's the only other spot it's used)